Automattic/wp-calypso
# Domains: Status for transfer out

#### Changes proposed in this Pull Request

This PR adds the transfer out status to the registered domain card.

This is how it appears for the domain owner:

<img width="726" alt="Screenshot 2020-04-09 at 3 32 22 PM" src="https://user-images.githubusercontent.com/13062352/78900782-9eac2480-7a77-11ea-915a-f328ba6a1e10.png">

Whereas this is how it appears for other users:

<img width="726" alt="Screenshot 2020-04-09 at 3 35 49 PM" src="https://user-images.githubusercontent.com/13062352/78900883-cef3c300-7a77-11ea-8378-19e16b15057f.png">

Depends on D41813-code

#### Testing instructions

- Either start a transfer out, or hack the backend to fake a transfer out status 
- Make sure the view appears as it should, for the owner and for other users
- Accept/cancel buttons should work as expected
- Assert that only the Cancel button shows when accepting the transfer is not supported
